From: SeongJae Park <sj@kernel.org>
Date: Mon, 24 Feb 2025 13:15:41 -0800
Subject: [PATCH] selftests/damon/damon_nr_regions: set ops update for merge
 results check to 100ms

After allowing sampling time parameters commit, damon_nr_regions
selftest frequently fails.  It is because the merging is limited by
sz_limit internal variable, which is updated only once per ops update
interval.  Sampling time commit makes the code easier to pass faster
without encountering the ops update operation, and hence sz_limit is not
properly updated.

Fix the issue by setting the ops interval same to aggregation.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 tools/testing/selftests/damon/damon_nr_regions.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/testing/selftests/damon/damon_nr_regions.py b/tools/testing/selftests/damon/damon_nr_regions.py
index 2e8a74aff543..6f1c1d88e309 100755
--- a/tools/testing/selftests/damon/damon_nr_regions.py
+++ b/tools/testing/selftests/damon/damon_nr_regions.py
@@ -109,6 +109,7 @@ def main():
     attrs = kdamonds.kdamonds[0].contexts[0].monitoring_attrs
     attrs.min_nr_regions = 3
     attrs.max_nr_regions = 7
+    attrs.update_us = 100000
     err = kdamonds.kdamonds[0].commit()
     if err is not None:
         proc.terminate()
-- 
2.39.5

